pkg/madmin/generic-commands.go

```diff
@@ -19,36 +19,33 @@ package madmin
 
 import (
 	"bytes"
-	"encoding/xml"
+	"encoding/json"
 	"net/http"
-	"net/url"
 )
 
-// setCredsReq - xml to send to the server to set new credentials
-type setCredsReq struct {
-	Username string `xml:"username"`
-	Password string `xml:"password"`
+// SetCredsReq - xml to send to the server to set new credentials
+type SetCredsReq struct {
+	AccessKey string `json:"accessKey"`
+	SecretKey string `json:"secretKey"`
 }
 
-// SetCredentials - Call Set Credentials API to set new access and secret keys in the specified Minio server
+// SetCredentials - Call Set Credentials API to set new access and
+// secret keys in the specified Minio server
 func (adm *AdminClient) SetCredentials(access, secret string) error {
-
-	// Setup new request
-	reqData := requestData{}
-	reqData.queryValues = make(url.Values)
-	reqData.queryValues.Set("service", "")
-	reqData.customHeaders = make(http.Header)
-	reqData.customHeaders.Set(minioAdminOpHeader, "set-credentials")
-
 	// Setup request's body
-	body, err := xml.Marshal(setCredsReq{Username: access, Password: secret})
+	body, err := json.Marshal(SetCredsReq{access, secret})
 	if err != nil {
 		return err
 	}
-	reqData.contentBody = bytes.NewReader(body)
-	reqData.contentLength = int64(len(body))
-	reqData.contentMD5Bytes = sumMD5(body)
-	reqData.contentSHA256Bytes = sum256(body)
+
+	// Setup new request
+	reqData := requestData{
+		relPath:            "/v1/config/credential",
```

Comment:
We should fail this operation if not on TLS connection, are we doing that?

Reply:
Fixed in both madmin and server handler to maximize chances that creds wont be sent over insecure connection. Good catch!

Reply:
TODO: Need to figure out how to fix the tests as a result of this change though.

Reply:
Fixed with a little hack.